libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteLineData.kt

```diff
@@ -1,43 +1,50 @@
 package com.mapbox.navigation.ui.maps.route.line.model
 
 import com.mapbox.geojson.FeatureCollection
+import com.mapbox.navigation.base.route.NavigationRoute
 
 /**
  * Provides information needed to draw a route.
  *
  * @param featureCollection the routes geometry
  * @param dynamicData dynamic data to style the route line
+ * @param navigationRoute [NavigationRoute] this route line corresponds to
  */
 class RouteLineData internal constructor(
     val featureCollection: FeatureCollection,
-    val dynamicData: RouteLineDynamicData
+    val dynamicData: RouteLineDynamicData,
+    val navigationRoute: NavigationRoute?,
```

Comment:
Why are you adding the NavigationRoute here?  This object shouldn't be in the view class.  It appears you're using it to get the route ID but the `Feature.id` of the `Feature` in the `FeatureCollection` is the route ID.  Look at MapboxRouteLineUtils::generateRouteFeatureData().